elastic/detection-rules
# Skip Development Rules from Security Docs

# Pull Request

*Issue link(s)*: https://github.com/elastic/detection-rules/issues/4008



## Summary - What I changed


- Added a skip criteria for Development rules in rule summary and rule reference. 
- The following rules are in development mode that need removal from security docs which was handled in separate docs PR

   - [Python Script Execution via Command Line](https://github.com/elastic/detection-rules/blob/main/rules/cross-platform/execution_python_script_in_cmdline.toml)
   - [Microsoft 365 Mass download by a single user](https://github.com/elastic/detection-rules/blob/main/rules/integrations/o365/exfiltration_microsoft_365_mass_download_by_a_single_user.toml)
   - [Microsoft 365 Impossible travel activity](https://github.com/elastic/detection-rules/blob/main/rules/integrations/o365/initial_access_microsoft_365_impossible_travel_activity.toml)
   - [Windows User Account Creation](https://github.com/elastic/detection-rules/blob/main/rules/windows/persistence_user_account_creation_event_logs.toml)
    - [WPAD Service Exploit](https://github.com/elastic/detection-rules/blob/main/rules/windows/privilege_escalation_wpad_exploitation.toml)

| Branch | PR | 
|--------|--------|
|8.12 | https://github.com/elastic/security-docs/pull/5828| 
| 8.13 |https://github.com/elastic/security-docs/pull/5826 | 
| 8.14 | https://github.com/elastic/security-docs/pull/5825 | 
| 8.15 | https://github.com/elastic/security-docs/pull/5824 | 

## How To Test



- We tested it on a tag that will generate fresh docs for example previous version was 8.14.6 and next version was 8.15.1
- Doc PR was raised to check the build and doc simulation - https://github.com/elastic/security-docs/pull/5831/
- The PR was checked to verify that the below rule files were not generated
    - microsoft-365-impossible-travel-activity.asciidoc
    - microsoft-365-mass-download-by-a-single-user.asciidoc
    - python-script-execution-via-command-line.asciidoc
    - wpad-service-exploit.asciidoc
    - windows-user-account-creation.asciidoc[]
- While development debug statements were added to ensure the rules are skipped 
<details><summary>Debug Details</summary>
<p>

```console
python -m detection_rules dev build-integration-docs 8.15.1 --pre 8.14.6 --post 8.15.1 -d ../security-docs \
    --update-message "Test" --force
Loaded config file: /Users/shashankks/elastic_workspace/detection-rules/.detection-rules-cfg.json

█▀▀▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄   ▄      █▀▀▄ ▄  ▄ ▄   ▄▄▄ ▄▄▄
█  █ █▄▄  █  █▄▄ █    █   █  █ █ █▀▄ █      █▄▄▀ █  █ █   █▄▄ █▄▄
█▄▄▀ █▄▄  █  █▄▄ █▄▄  █  ▄█▄ █▄█ █ ▀▄█      █ ▀▄ █▄▄█ █▄▄ █▄▄ ▄▄█

From github.com:elastic/detection-rules
 * tag                   integration-v8.14.6 -> FETCH_HEAD
From github.com:elastic/detection-rules
 * tag                   integration-v8.15.1 -> FETCH_HEAD
Skipping development rule Microsoft 365 Impossible travel activity for rule reference
Skipping development rule Microsoft 365 Mass download by a single user for rule reference
Skipping development rule Python Script Execution via Command Line for rule reference
Skipping development rule WPAD Service Exploit for rule reference
Skipping development rule Windows User Account Creation for rule reference
Generated documents saved to: ../security-docs/docs/detections/prebuilt-rules/downloadable-packages/8-15-1
- 72 updated rules
- 14 new rules
- 1 deprecated rules
(.venv) 
detection-rules on  issue-4008 [$!?] is 📦 v0.1.0 via 🐍 v3.12.5 (.venv) on ☁️  shashank.suryanarayana@elastic.co took 2m28s 
❯ 
```
</p>
</details> 

- Mock Build was completed for the private branch 
    - the failure is unrelated to the change, we have cut a branch from main and built an example for 8.15 and trying to merge to main.
    - So cross reference links are broken.
## Checklist




## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?
